tasks/task_database.c

```diff
@@ -145,47 +148,68 @@ static int task_database_iterate_start(retro_task_t *task,
    return 0;
 }
 
-static int intfstream_get_serial(intfstream_t *fd, char *serial)
+static int intfstream_get_serial(intfstream_t *fd, char *serial, const char *filename)
 {
-  const char *system_name = NULL;
-
-  /* Check if the system was not auto-detected. */
-  if (detect_system(fd, &system_name) < 0)
-  {
-    /* Attempt to read an ASCII serial, like Wii. */
-    if (detect_serial_ascii_game(fd, serial))
-    {
-      /* ASCII serial (Wii) was detected. */
-      RARCH_LOG("%s '%s'\n", msg_hash_to_str(MSG_FOUND_DISK_LABEL), serial);
-      return 0;
-    }
-
-    /* Any other non-system specific detection methods? */
-    return 0;
-  }
-
-  if (string_is_equal(system_name, "psp"))
-  {
-    if (detect_psp_game(fd, serial) == 0)
-      return 0;
-    RARCH_LOG("%s '%s'\n", msg_hash_to_str(MSG_FOUND_DISK_LABEL), serial);
-  }
-  else if (string_is_equal(system_name, "ps1"))
-  {
-    if (detect_ps1_game(fd, serial) == 0)
-      return 0;
-    RARCH_LOG("%s '%s'\n", msg_hash_to_str(MSG_FOUND_DISK_LABEL), serial);
-  }
-  else if (string_is_equal(system_name, "gc"))
-  {
-    if (detect_gc_game(fd, serial) == 0)
+   const char *system_name = NULL;
+   if (detect_system(fd, &system_name, filename) >= 1)
+   {
+      /** Debug: runloop_msg_queue_push(system_name, 0, 180, false, NULL, MESSAGE_QUEUE_ICON_DEFAULT, MESSAGE_QUEUE_CATEGORY_INFO); **/
```

Comment:
> Debug message? We could stick this into the log with RARCH_LOG.

It exists in RARCH_LOG.  This code could be removed, it will popup the serial and name of game onscreen in the flash bars.

Reply:
> > Debug message? We could stick this into the log with RARCH_LOG.
> 
> It exists in RARCH_LOG. This code could be removed, it will popup the serial and name of game onscreen in the flash bars.

Removed this debug code.